lib/type1.g

```diff
@@ -294,9 +333,16 @@ BIND_GLOBAL( "NEW_TYPE", function ( typeOfTypes, family, flags, data, parent )
         cache[hash] := type;
     fi;
     family!.nTYPES := family!.nTYPES + 1;
+    if IsHPCGAP then
+        MakeReadOnlySingleObj(type);
+        UNLOCK(lock);
+    fi;
 
     # return the type
     return type;
+    if not IsHPCGAP then
+        return lock; # HACK to silence C compiler warnings on generated code
```

Comment:
I'd like to draw attention to this nasty HACK I added to silence warnings by the C compiler about unused variables in the C code generated by `gac`. If somebody has a better idea...

Reply:
We could make the compiler output `(void)local;` for all local variables, which is (in my experience) the "standard" way of forcing a variable to look used, but that would be a bigger change.

Reply:
Of course adding support for `atomic` to `gac` would allow us to change this code back to using `atomic`, which would neatly avoid the need for this HACK.

Reply:
This would be easy(ish) to do, if we required a compiler with the __cleanup__ extension, which includes gcc and clang. HPC-GAP doesn't (I imagine) work on anything but a recent gcc or clang anyway.

Reply:
Or, even more shocking, make the compiler output C++ (perhaps just for HPC-GAP at first)

Reply:
I am very reluctant to make us rely strongly on a non-standard compiler extension, and don't like the idea of locking us in to GCC and clang (sure, that looks fine now, but in the future...?) Using C++ is more attractive (for reasons discussed before), though so far nobody has invested time in making that possible (it requires some tedious work).

But anyway: All in all, correctly implementing support for `atomic` in gac using plain C is not *that* hard. I am sure I could do it in a day or two -- but I lack motivation (i.e. the gain is minimal, and HPC-GAP currently doens't work very well for me anyway).

Reply:
Anyway, discussion of atomic support in gac should go to issue #2108

Reply:
Anyway, to get back to the point: I could indeed emit `(void)l_unused_local`, it's a tiny change, and doesn't do much in the `src/c_*.c` files. If people are OK with it, I'll augment this PR accordingly.